src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -988,9 +988,20 @@ private HttpRequestMessage CreateRequest<TResult>(JsonOperationContext ctx, Serv
 
             request.RequestUri = new Uri(url);
 
-            if (!request.Headers.Contains(Constants.Headers.ClientVersion))
+            if (request.Headers.Contains(Constants.Headers.ClientVersion) == false)
                 request.Headers.Add(Constants.Headers.ClientVersion, ClientVersion);
 
+            if (command.IsClusterCommand)
+            {
+                if (command.Guid == null)
+                {
+                    command.Guid = Guid.NewGuid().ToString();
+                }
+
+                if (request.Headers.Contains(Constants.Headers.RaftCommandGuid) == false)
```

Comment:
That isn't a good idea. We need to have the Guid directly in the command json itelf.
Certain commands may do a wire hop. Go over from node A to node B (which is the leader) and we want to Guid to be part of that.
For that matter, not crazy about the `Guid` as a name. A UniqueCommandId is better

Reply:
The reason I've passed it via the headers is because it is easier to append it on the request.
Wasn't aware that redirecting doesn't preserve the headers...

Reply:
Putting it in header / url means that it is outside of the data itself. We need it _in_ the command.